.clang-tidy

```diff
@@ -12,13 +12,21 @@ Checks: |
   -cppcoreguidelines-pro-bounds-pointer-arithmetic
   -cppcoreguidelines-init-variables
   -cppcoreguidelines-pro-type-member-init
+  -cppcoreguidelines-avoid-do-while
```

Comment:
I know I saw these once but don't see them in the latest [clang-tidy output from develop](https://productionresultssa4.blob.core.windows.net/actions-results/1e27f446-2749-4de8-934c-49550066eb6b/workflow-job-run-0f44d42c-80fb-5772-bde9-fe33f2c238e7/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-08-05T11%3A39%3A37Z&sig=Q%2FXD0qidw1S%2B4D770tQPN7ZxQpV%2FtCB3xtZO9WedILE%3D&ske=2024-08-05T19%3A29%3A48Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-08-05T07%3A29%3A48Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-05-04&sp=r&spr=https&sr=b&st=2024-08-05T11%3A29%3A32Z&sv=2024-05-04)...

Reply:
Mmh right, I'm not sure none is showing up, I see it locally: 
```
[144/338] Building CXX object src/celeritas/CMakeFiles/celeritas_hepmc.dir/io/EventReader.cc.o
/home/esseivaj.linux/devel/celeritas/src/celeritas/io/EventReader.cc:53:9: warning: avoid do-while loops [cppcoreguidelines-avoid-do-while]
   53 |         do
```

I'm using clang 17 so this might explain the differences.